src/app/providers.tsx

```diff
@@ -26,14 +27,16 @@ export function Providers({ children }: { children: React.ReactNode }) {
       <ChakraProvider theme={theme}>
         <ColorModeScript initialColorMode={theme.config.initialColorMode} />
         <WagmiProvider config={wagmiConfig}>
-          <QueryClientProvider client={queryClient}>
-            <RainbowKitProvider
-              theme={rainbowkitTheme}
-              appInfo={rainbowKitAppInfo}
-            >
-              <AnalyticsProvider>{children}</AnalyticsProvider>
-            </RainbowKitProvider>
-          </QueryClientProvider>
+          <SignTermsProvider>
+            <QueryClientProvider client={queryClient}>
+              <RainbowKitProvider
+                theme={rainbowkitTheme}
+                appInfo={rainbowKitAppInfo}
+              >
+                <AnalyticsProvider>{children}</AnalyticsProvider>
+              </RainbowKitProvider>
+            </QueryClientProvider>
+          </SignTermsProvider>
```

Comment:
Wouldn't it be better to place this inside QueryClient and RainbowKit? So just wrapping the AnalyticsProvider.

Reply:
Or maybe even inside that one in case you wanna use logic from there.

Reply:
@janndriessen good call - done 👍